### PR TITLE
feat: fork mode features and traces

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What I did

fixes: #6

* docs: correct things in readme
* fix: Use new API method for getting virtual_machine_error and fix bugs related to revert message detection
* feat: add ability to set request timeouts
* fix: better connection-ready detection
* feat: ability to get traces. Requires: https://github.com/ApeWorX/evm-trace/pull/7

**NOTE**: Still having issue with Anvil regarding process-drops and forked network bugs.

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

**Comment by banteg** [comment]
Can confirm account impersonation works properly with `anvil` from https://github.com/foundry-rs/foundry/pull/1912

You can install it with `foundryup --pr 1912`

**Comment by unparalleled-js** [comment]
We need this issue resolved https://github.com/foundry-rs/foundry/issues/1871
It was closed but I am still having problems (see comments)

**Comment by unparalleled-js** [comment]
Remaining Anvil PRs needed for this plugin to work:

- [ ] https://github.com/foundry-rs/foundry/issues/2036

**Comment by unparalleled-js** [comment]
Fork mode still does not seem to be functioning correctly.

Transaction do not work when they are supposed.
And then also sometimes they do work when they are not supposed to!

The normal local provider is working great though :)

**Comment by unparalleled-js** [comment]
- [ ] https://github.com/foundry-rs/foundry/issues/2036

**Review by fubuloubu — APPROVED** [review]

File: tests/conftest.py [hunk 1]
```diff
@@ -1,70 +1,147 @@
...
-from ape_foundry import FoundryProvider
+from ape_foundry import FoundryForkProvider, FoundryProvider
 
+# NOTE: Ensure that we don't use local paths for the DATA FOLDER
+ape.config.DATA_FOLDER = Path(mkdtemp()).resolve()
 
-def get_project() -> ProjectManager:
-    return Project(Path(__file__).parent)
+BASE_CONTRACTS_PATH = Path(__file__).parent / "data" / "contracts"
+logger.set_level(LogLevel.DEBUG)
```
Comment (unparalleled-js) [inline tests/conftest.py:18]: can get rid of this once https://github.com/ApeWorX/ape/pull/889 merges and is released and usable

**Review by fubuloubu — APPROVED** [review]

**Review by fubuloubu — APPROVED** [review]

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**